src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -248,6 +248,12 @@ protected function getMigrationsForRollback(array $options)
             return $this->repository->getMigrations($steps);
         }
 
+        if ($path = $options['path'] ?? []) {
+            return collect($path)->map(function ($m) {
+                return (object) ['migration' => (string) \Str::of($m)->basename('.php')];
```

Comment:
`use Illuminate\Support\Str;` is already defined in this file so `Str::of($m)->basename('.php')` without `\` is OK.

This also wouldn't be able to assume root facade `Str`  is defined since upgraded apps made not have added it or userland may have decided to remove some aliases:

https://github.com/laravel/laravel/blob/399d435c4f0b41a5b6d3e14894195f9196d36bb8/config/app.php#L228

Reply:
```suggestion
                return (object) ['migration' => (string) Str::of($m)->basename('.php')];
```

Reply:
Fixed, thank you @derekmd.